### ISSUE TITLE
[P2][dogfood] Two dependency surfaces disagree on circular-dependency count (--dependencies=16 vs --import-graph=8) on identical graph state

### ISSUE THREAD

**Issue by aimasteracc** [aimasteracc/tree-sitter-analyzer#784]
**Severity:** P2  |  **Surface:** dependency analysis cycle counting (--dependencies vs --import-graph)
_Found by parallel dogfood audit round 2 on develop @ 471ac5f5; adversarially re-reproduced + dup-checked by an independent verifier._

### Observed
--import-graph reports cycle_count=8 (summary and cycles modes agree; enumerates 8 full paths). --dependencies summary reports cycle_count=16 with summary_line "summary: 1930 nodes / 4075 edges / 16 cycles", and --dependencies cycles enumerates exactly 16 paths. Same project, same warm cache, identical node/edge counts (1930/4075) — a 2x difference in cycle count between the two overlapping dependency tools. The extra 8 in --dependencies come from benchmarks/, tests/fixtures/, single-file __init__.py self-loops, and plugin-package __init__ re-export cycles that --import-graph omits. (The original report's claim that --dependencies crashes except on a cold run was NOT reproduced; it ran cleanly warm.)

### Expected
Both views are presented as project module/import dependency analysis (--import-graph is even labeled "CodeGraph parity"); they should report the same cycle_count on the same graph, OR --help / the output should document why the graph models differ (e.g. file-scoping rules, inclusion of benchmarks/tests/fixtures, treatment of single-file __init__.py self-loops and package re-export cycles). A consumer cannot trust a cycle_count that silently depends on which of two overlapping tools they happened to call.

### Repro
```bash
cd /Users/aisheng.yu/git-private/tsa-dogfood (develop @471ac5f5)
1) uv run python -m tree_sitter_analyzer --import-graph --format json | python3 -c "import json,sys;print(json.load(sys.stdin)['cycle_count'])"  -> 8
2) uv run python -m tree_sitter_analyzer --dependencies summary --format json | python3 -c "import json,sys;d=json.load(sys.stdin);print(d['cycle_count'],d['summary_line'])"  -> 16 'summary: 1930 nodes / 4075 edges / 16 cycles'
3) Cross-check full enumeration: --import-graph --import-graph-mode cycles -> 8 paths; --dependencies cycles -> 16 paths.
Both report node_count=1930 / edge_count=4075, so it is the same graph state.
```

### Verifier notes
Reproduced deterministically on develop @471ac5f5, warm cache, identical graph state (both surfaces report node_count=1930 / edge_count=4075). `--import-graph --format json` -> cycle_count=8; `--dependencies summary --format json` -> cycle_count=16 with summary_line "summary: 1930 nodes / 4075 edges / 16 cycles". Confirmed across modes: `--import-graph --import-graph-mode cycles` enumerates exactly 8 full paths; `--dependencies cycles` enumerates exactly 16. Not a cold-vs-warm artifact (both warm, same counts), not a flag/user error (both are documented project-level dependency flags). The divergence is a genuine graph-model difference, not noise: the `--dependencies` set additionally includes benchmarks/ cycles (3), tests/fixtures/ cycles (1), single-file __init__.py self-loops (e.g. synapse_resolver/__init__.py -> __init__.py), and plugin-package __init__ re-export cycles (markdown/sql/typescript) that `--import-graph` excludes or scopes out. Both are deterministic and fully enumerable, so a consumer comparing the two trustworthy-looking dependency tools gets a 2x discrepancy with no explanation in --help. Note: the original observed text mentioned --dependencies crashing on warm runs / only succeeding cold — I did NOT reproduce any crash (it ran cleanly on warm cache every time); only the 16-vs-8 count divergence is confirmed. No overlap with open issues #734-757 (call-path, symbol-search caps, JS/Go/C/C++ extraction, structure flags, envelope omissions, smart-context, check-scale, symbol-lineage) or any other open issue.

**Comment by aimasteracc** [comment]
Fixed in PR #896 (merged). scope + scope_note added to dependency summary mode.
